src/rgw/rgw_op.cc

```diff
@@ -1045,13 +1045,10 @@ int handle_cloudtier_obj(req_state* s, const DoutPrefixProvider *dpp, rgw::sal::
         ldpp_dout(dpp, 5) << "restore is not completed yet, please retry again" << dendl;
         s->err.message = "restore is not completed yet";
       }
-      return op_ret;
     } else if (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
       op_ret = -ERR_INVALID_OBJECT_STATE;
       ldpp_dout(dpp, 5) << "restore is not completed yet, please retry again" << dendl;
       s->err.message = "restore is not completed yet";
-    } else { // CloudRestored..return success
```

Comment:
I don't have any objection but I would still prefer to keep these else parts for clear understanding when looking at the code .. it ensures all cases are handled.